src/ray/core_worker/reference_count.cc

```diff
@@ -247,6 +247,19 @@ void ReferenceCounter::SetNestedRefInUseRecursive(ReferenceTable::iterator inner
   }
 }
 
+void ReferenceCounter::ReleaseAllLocalReferences() {
+  ReferenceTable all_local_refs;
+  {
+    absl::MutexLock lock(&mutex_);
+    /// A copy of object_id_refs_ to avoid iter invalidation
+    /// and mutex lock reentred.
+    all_local_refs = object_id_refs_;
+  }
+  for (auto &ref : all_local_refs) {
+    RemoveLocalReference(ref.first, nullptr);
```

Comment:
I think we need to call this multiple times, if the worker has multiple instances of the object.

You can test this in Python by passing the same object as two of the task's arguments.

Reply:
Good catch.